numba/cgutils.py

```diff
@@ -676,6 +676,16 @@ def get_item_pointer2(builder, data, shape, strides, layout, inds,
             indices.append(selected)
     else:
         indices = inds
+    if boundscheck:
+        for ind, dimlen in zip(indices, shape):
+            msg = "index is out of bounds"
+            out_of_bounds_upper = builder.icmp_signed('>=', ind, dimlen)
```

Comment:
I don't know if it matters if this is signed or unsigned. Both seem to work.

Reply:
I think it needs to be signed, though wonder in practice if this case would arise as `wraparound` would most likely be set to produce a normalized index.

Reply:
Oh I hadn't considered if wraparound was False.